pkg/storage/storage.go

```diff
@@ -163,6 +164,49 @@ func New(cfg *config.Config) (*Storage, error) {
 		return tree.New()
 	}
 
+	// load the total memory of the server
+	vm, err := mem.VirtualMemory()
+	if err != nil {
+		return nil, err
+	}
+
+	interval := time.Duration(10)
+	// start a timer for checking if the memory used by application
+	// is more than 25% of the total memory, if so, trigger eviction
+	// with 10% to every cache
+	go func() {
+		ticker := time.NewTimer(interval * time.Second)
```

Comment:
I think a `Ticker` is more appropriate here.

Reply:
why?

Reply:
* [The `Timer` type represents a single event](https://golang.org/src/time/sleep.go?s=1658:1710#L40)
 * [A `Ticker` holds a channel that delivers “ticks” of a clock at intervals](https://golang.org/src/time/tick.go?s=271:373#L1)

Timers are for when you want to do something once in the future - tickers are for when you want to do something repeatedly at regular intervals.

Reply:
yes, but i reset the timer in the loop

// reset the timer
ticker.Reset(interval * time.Second)

Reply:
For sure that works. However, I think it would be better to use more idiomatic approach with `Ticker` :)

Reply:
@kolesnikovae I talked with @alonlong about it. I think the idea is that if this procedure takes longer than the timer we want to wait for another X seconds (10 right now I think) before flushing.

So I think it's fine. Although, I wonder if we should just add a time.Sleep there. What do you both think about that? I think that would make code more concise, but maybe there're some downsides I'm not thinking of.

Reply:
`Ticker` also has `Reset` method for that.

Sorry guys if led you astray. Please don't get me wrong, I was just trying to say that `Timer` and `Ticker` have different semantics, and for this particular case the last one is more appropriate IMO. Taking into account the mismatch of the variable name and the type it's a bit confusing: is it supposed to be a timer and the variable name should be changed, or is it a ticker indeed and the type should be changed? Both options are okay.

Reply:
@kolesnikovae Got it, I think we can change variable name to timer